cpp/include/cuvs/cluster/kmeans.hpp

```diff
@@ -781,9 +782,53 @@ void predict(const raft::resources& handle,
  */
 void predict(const raft::resources& handle,
              cuvs::cluster::kmeans::balanced_params const& params,
-             raft::device_matrix_view<const float, int> X,
-             raft::device_matrix_view<const float, int> centroids,
-             raft::device_vector_view<int, int> labels);
+             raft::device_matrix_view<const int8_t, int64_t> X,
+             raft::device_matrix_view<const float, int64_t> centroids,
+             raft::device_vector_view<int, int64_t> labels);
+
+/**
+ * @brief Predict the closest cluster each sample in X belongs to.
+ *
+ * @code{.cpp}
+ *   #include <raft/core/resources.hpp>
+ *   #include <cuvs/cluster/kmeans.hpp>
+ *   using namespace  cuvs::cluster;
+ *   ...
+ *   raft::resources handle;
+ *   cuvs::cluster::kmeans::balanced_params params;
+ *   int64_t n_features = 15, n_clusters = 8;
+ *   auto centroids = raft::make_device_matrix<float, int64_t>(handle, n_clusters, n_features);
+ *
+ *   kmeans::fit(handle,
+ *               params,
+ *               X,
+ *               centroids.view());
+ *   ...
+ *   auto labels = raft::make_device_vector<int, int64_t>(handle, X.extent(0));
+ *
+ *   kmeans::predict(handle,
+ *                   params,
+ *                   X,
+ *                   centroids.view(),
+ *                   labels.view());
+ * @endcode
+ *
+ * @param[in]     handle           The raft handle.
+ * @param[in]     params           Parameters for KMeans model.
+ * @param[in]     X                New data to predict.
+ *                                 [dim = n_samples x n_features]
+ * @param[in]     centroids        Cluster centroids. The data must be in
+ *                                 row-major format.
+ *                                 [dim = n_clusters x n_features]
+ * @param[out]    labels           Index of the cluster each sample in X
+ *                                 belongs to.
+ *                                 [len = n_samples]
+ */
+void predict(const raft::resources& handle,
+             cuvs::cluster::kmeans::balanced_params const& params,
+             raft::device_matrix_view<const float, int64_t> X,
+             raft::device_matrix_view<const float, int64_t> centroids,
+             raft::device_vector_view<int, int64_t> labels);
```

Comment:
Shouldn't the labels type and extent be of the same type? By allowing int64_t extents, labels can also go beyond int32 limits

Reply:
Thank you for the comment. Yes, it can exceed the int32 limits. If it would be better to add the `int64_t` support for the label type, I'll do so. The original purpose of this PR was to pass the `RAFT_EXPECTS` by changing `IndexT` from `int` to `int64_t`, so adding the support is not mandatory.
https://github.com/rapidsai/cuvs/blob/50ed7109abc2bb579e323404a6a7525adc7da275/cpp/src/cluster/kmeans_balanced.cuh#L86-L88